subsys/bluetooth/mesh/gen_prop_srv.c

```diff
@@ -46,23 +50,39 @@ static struct bt_mesh_prop *prop_get(const struct bt_mesh_prop_srv *srv,
 	return NULL;
 }
 
-static void store_props(const struct bt_mesh_prop_srv *srv)
+#if CONFIG_BT_SETTINGS
+static void store_timeout(struct k_work *work)
 {
-	if (!IS_ENABLED(CONFIG_BT_SETTINGS)) {
-		return;
-	}
+	int err;
+
+	struct bt_mesh_prop_srv *srv = CONTAINER_OF(
+		work, struct bt_mesh_prop_srv, store_timer.work);
```

Comment:
This is not really correct, the `work` parameter is `srv->work.work`. Should use `k_work_delayable_from_work()`

Reply:
I believe this is correct. I have been checking in light_ctrl_srv and it is done this way both for action_delay and reg. Is not enough to have work as input when having store_timer.work. I this is not true I should change all the models.

Reply:
Sounds like we should change this in all models then, because it's not correct. Created NCSDK-9807 for this.

Reply:
Discussed offline. New code should use `k_work_delayable_from_work()`. The current implementation will be fixed in #4725

Reply:
Fixed